### DISCUSSION TITLE
Workout with ERG off

### DISCUSSION THREAD

**Discussion by TClin76** [dvmarinoff/Auuki#117] [Ideas]
Hi @dvmarinoff 
Is there anyway to do workout without ERG control by Flux?
If smart trainer is allowed to be paired under "POWER METER" in flux, instead of "CONTROLLABLE", that should work.
Thanks

**Comment by dvmarinoff** [comment]
If the trainer is implementing and advertasing the power meter BLE spec, it should show in the search popup for power meter. You will need to switch the source for power from controllable to power meter from the menu in settings.

If this does not work for you let me know with more details on your particular use case.

**Reply by TClin76** [reply]
> If the trainer is implementing and advertasing the power meter BLE spec, it should show in the search popup for power meter. You will need to switch the source for power from controllable to power meter from the menu in settings.
> 
> If this does not work for you let me know with more details on your particular use case.

I tried this method before, but it didn't show up in popup window of Power meter.
By the way, my trainer is Magene GRAVAT.

I believe that training without ERG brings you additional benefit of cadence control ability. 
It would be nice to have an ERG switch function in Flux, if there is an easy way to do it.
Anyway, it is a minor suggestion, not worth spending too much time to implement this function.

**Reply by dvmarinoff** [reply]
Would locking the workout into Slope mode work for your use case?

I am a bit confused by:
> training without ERG brings you additional benefit of cadence control ability

**Reply by TClin76** [reply]
Locking into slope mode during the whole workout should work, I think. 
The current behavior in Flux is that slope mode will switch back to ERG mode when new interval starts.
But again it is not a very important feature for Flux, so please not to spend too much time on it. Thanks.

Regarding to non-ERG training:
I have trained with ERG for years. Some coaches encouraged me try to do the workout without the help of ERG.
I found it really helped for improving my ability to maintain higher cadence in Z4 or even higher zone.
Frankly speaking I am not so sure how it makes a difference but it has definitely made me progress.

Or below link may help to explain.
https://www.360velo.com/why-dont-use-erg-mode/

**Reply by dvmarinoff** [reply]
OK, I'll try to find solution for locking the mode. The main difficulty is that the App uses custom events and the UI input and the workout commands use the same set of events. Will need to decouple them without braking the already complex relationship between the modes and the type of rides (workouts, courses).

**Reply by TClin76** [reply]
How about implementing in this way?
1. Whenever user manually click "resistance" or "slope" during workout, it will decouple the ERG control till the end of workout. 
In this case, Flux only resumes ERG mode when user manually click "erg".

2. If Flux switch to RESISTANCE mode due to the interval command "Freeride" in .zwo, it will automatically switch back to ERG mode while leaving Freeride interval.

3. If the type of ride is "Courses", it will always stay in SLOPE mode.

**Reply by dvmarinoff** [reply]
Having all modes in Course rides is important feature for me.

It's implementation issue right now. I can go in 2 ways to implement the lock:
1. introduce different events for user input, which will always overwrite as priority the workout/course commands,
2. add a lock button that will introduce a lock mode variable for the existing events,


1 would require changes to multiple files with critical functionality, 2 is easier to implement, but would require additional UI, hence more clutter.

**Reply by TClin76** [reply]
How about this:
No need to implement 3 (from my last reply)
1 & 2 (from my last reply) are only worked when user select a .zwo workout, but not a .fit course.

So that you don't have to change UI and also no need to introduce other user input file.
